src/displayapp/screens/Dice.cpp

```diff
@@ -0,0 +1,231 @@
+  diceSizeIncBtnLabel = lv_label_create(diceSizeIncBtn, nullptr);
+  lv_label_set_text(diceSizeIncBtnLabel, "+");
+
+  diceSizeLabelCont = lv_cont_create(configCont, nullptr);
+  lv_obj_set_size(diceSizeLabelCont, 80, 40);
+  lv_obj_set_pos(diceSizeLabelCont, 80, 60);
+  lv_obj_set_style_local_bg_color(diceSizeLabelCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_cont_set_layout(diceSizeLabelCont, LV_LAYOUT_CENTER);
+  diceSizeLabel = lv_label_create(diceSizeLabelCont, nullptr);
+  lv_label_set_text_fmt(diceSizeLabel, "d%d", 6);
+
+  diceSizeDecBtn = lv_btn_create(configCont, nullptr);
+  diceSizeDecBtn->user_data = this;
+  lv_obj_set_event_cb(diceSizeDecBtn, dice_size_dec_event_handler);
+  lv_obj_set_size(diceSizeDecBtn, 70, 50);
+  lv_obj_set_pos(diceSizeDecBtn, 85, 105);
+  lv_obj_set_ext_click_area(diceSizeDecBtn, 5, 5, 10, 5);
+  diceSizeDecBtnLabel = lv_label_create(diceSizeDecBtn, nullptr);
+  lv_label_set_text(diceSizeDecBtnLabel, "-");
+}
+
+void Dice::rollDice() {
+  uint8_t roll = 0;
+  uint16_t result = 0;
+
+  lv_label_set_text(resultTotalLabel, "");
+  lv_label_set_text(resultIndRollsLabel, "");
+
+  for (int i = 0; i < numDice; i++) {
+    roll = rand() % diceSizes[diceSizeIndex] + 1;
+    result += roll;
+
+    if (numDice > 1) {
+      if (i == 0) {
+        lv_label_ins_text(resultIndRollsLabel, 0, "(");
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, std::to_string(roll).c_str());
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, ",");
+      } else if (i < numDice - 1) {
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, std::to_string(roll).c_str());
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, ",");
+      } else {
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, std::to_string(roll).c_str());
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, ")");
+      }
+    }
+  }
+
+  lv_label_set_text_fmt(resultTotalLabel, "%d", result);
+}
+
+Dice::Dice(DisplayApp* app,
+           Controllers::MotionController& motionController,
+           Controllers::MotorController& motorController,
+           System::SystemTask& systemTask)
+  : Screen(app),
+    systemTask {systemTask},
+    motionController {motionController},
+    motorController {motorController},
+    numDice(1),
+    diceSizeIndex(3) {
+
+  app->SetTouchMode(DisplayApp::TouchModes::Polling);
+  createScreen();
+}
+
+Dice::~Dice() {
+  lv_obj_clean(lv_scr_act());
+  app->SetTouchMode(DisplayApp::TouchModes::Gestures);
+}
+
+bool Dice::Refresh() {
+  if (motionController.ShouldRollDice()) {
+    motorController.SetDuration(60);
+    rollDice();
+  }
+  return running;
+}
+
+bool Dice::OnTouchEvent(Pinetime::Applications::TouchEvents event) {
+  return true;
+}
+
+void Dice::numDiceIncBtnEventHandler(lv_event_t event) {
+  if (event != LV_EVENT_CLICKED) {
+    return;
+  }
+
+  if (numDice < 99) {
```

Comment:
Such high number is nice, but can You see all 99 dice results in the app? I think something like 10 would be enough

Reply:
You can certainly see the total -- and the individual rolls label uses the circular scroll long mode, so *technically* you could see all of the individual rolls, though it's unlikely somebody would wait all the way through that.

10 is definitely a bit low, but I'd be happy to lower it to something else if needed.  At 99 100-sided dice it's noticeably slower on the display, but it still handles it mostly fine.